Sources/FoundationExtensions/URL.swift

```diff
@@ -7,3 +7,9 @@ extension URL {
         return o
     }
 }
+
+extension String {
+    public var url: URL? {
+        return URL(string: self)
```

Comment:
A number of times I ended up regretting this sort of computed property and wished I made them throwing methods instead. Those can be still chained and optionalised, but they can also be super useful in system tracking and debugging where they are allowed to throw. A method could also take a defaulting base url argument and any options (like whether this is a file reference and such)... What do you think?

Reply:
Would it be wise to have both options available? It seems to me that the computed properties are very useful because they can be accessed via keypath

Reply:
Maybe thats a bad idea actually. It would balloon the number of accessors we'd be implementing

Reply:
For what its worth, The implementation for keyPaths to instance functions is currently being worked on https://github.com/pcantrell/swift-evolution/blob/method-keypaths/proposals/NNNN-method-keypaths.md

Reply:
Yep! Can't wait! I'll happily work around the key path limitation if we do decide that throwing and optionalised functions are the way to go

Reply:
However Ste decides to resolve this, that was an awesome thing to draw attention to Noah 👍